### ISSUE TITLE
ncp-web/wizard wishlist

### ISSUE THREAD

**Issue by nachoparker** [nextcloud/nextcloudpi#622]
Wishlist of features / enhancements for ncp-web. This list complements #380, but these mostly don't need to wait until the backend changes.

## ncp-web

- [ ] a special view for nc-restore where you can see a list of available backups and click on one of them
- [ ] a directory picker widget so you can select a folder by browsing the server, instead of typing the path. This would be useful for several apps, like nc-datadir, nc-database...
- [ ] a search box
- [ ] a way to upload files, such as a backup or a config file
- [ ] ability to show/hide the sidebar with a swipe (mobile phone)
- [ ] sanitize input in javascript. Probably after #380
- [x] associate urls with sections, and be able to use history ( eg: `nextcloudpi.local:4443/#nc-datadir`)
- [x] file exists indication when there is a path in the text box
- [ ] making the translation "engine" a bit easier to use for translators (just use shortcodes instead of full strings to match)
- [x] a python test in the `test` folder
- [x] "typed fields", so for instance passwords are hidden while typing, checkboxes, file paths...

## wizard

- [ ] the wizard needs a couple new things integrated: enable automatic NC updates and NCP updates, maybe activate UFW
- [ ] rewarn before formatting that the contents of the drive will be lost
- [ ] check for updates/update at beginning of wizard (like ncp-web)
- [ ] disable fail2ban,dnsmasq if selecting "no external access"?
- [ ] more DDNS clients
- [ ] prefill text input boxes with past input
- [ ] detect open ports, and only ask to use upnp if needed
- [ ] detect nat loopback and only activate dnsmasq if necessary
- [ ] new proposed design in https://github.com/nextcloud/nextcloudpi/issues/272
- [ ] translations

**Comment by chiyi4488** [comment]
I think maybe can add a language setting to the wizard.
and add translate in wizard.

**Comment by prog-amateur** [comment]
About  `"typed fields", so for instance passwords are hidden while typing, checkboxes, file paths...`

I am not a developer, but I hope this can help. On the website _Nextcloud demo_, I could find both user field and password field code : 

**User ID field :** 
```
<p class="grouptop">
			<input name="user" id="user" placeholder="Username or email" aria-label="Username or email" value="" autofocus="" autocomplete="on" autocapitalize="none" autocorrect="off" required="" data-cip-id="user" type="text">
			<label for="user" class="infield">Username or email</label>
</p>
```

**Password field :** 
```
<p class="groupbottom">
			<input name="password" id="password" value="" placeholder="Password" aria-label="Password" autocomplete="on" autocapitalize="off" autocorrect="none" required="" data-cip-id="password" type="password">
			<label for="password" class="infield">Password</label>
</p>
```
Maybe this can help to solve this point by adjusting NCP code ? Thank you.

**Comment by nachoparker** [comment]
Thanks @prog-amateur. The issue is that right now all fields are just text. We need to differentiate which ones are passwords.

**Comment by chiyi4488** [comment]
I think some passwords don't need to be hidden.Because the passwords generated automatically by the system are very long, if you forget them, people can also recover them from ncp-web or config.
Of course, if you want to take security first,  still have to make password hiding.
Or more to restore the password function

**Comment by nachoparker** [comment]
You can always restore your password from SSH or by plugging in a terminal and a keyboard and running `ncp-config` > `nc-passwd`

**Comment by chiyi4488** [comment]
Oh, I forget it. Sorry.

**Comment by loveboylion** [comment]
I am not sure if its intentional or a bug but its related to setup wizard so posting it here screenshot tells all 
https://ibb.co/fa8DeK
nc-data dir refusing to move data

**Comment by sakura-lain** [comment]
Hi,
I'm sorry if I missed it, I made some researche but didn't find the solution : it would be nice if the webUI user name could be changed. Only the password can.

Thanks!

**Comment by Zakspeed** [comment]
In View System Info: 
Temp micro: "watch -n 0,1 vcgencmd measure_temp"

**Comment by Hofer-Julian** [comment]
> I am not sure if its intentional or a bug but its related to setup wizard so posting it here screenshot tells all
> [![2018_09_29](https://camo.githubusercontent.com/6b5d5f822de15b3d7d1378bf65c94f38d13a40bb/68747470733a2f2f707265766965772e6962622e636f2f67386a6d7a4b2f323031385f30395f32392e706e67)](https://ibb.co/fa8DeK)
> nc-data dir refusing to move data
> also one more while updating nextcloud using gui
> [![2018_09_30](https://camo.githubusercontent.com/d756b1c1df88d9ef51437e309f9ec5a59455fbb3/68747470733a2f2f707265766965772e6962622e636f2f637730627a4b2f323031385f30395f33302e706e67)](https://ibb.co/nyrieK)
> clean setup on raspberry pi 1 in raspbian os fully upadted and as root user using below command
> # curl -sSL https://raw.githubusercontent.com/nextcloud/nextcloudpi/master/install.sh | bash

I would like to highlight this problem as well.

I think one way to circumvent this problem would be to simply change one line in https://github.com/nextcloud/nextcloudpi/blob/master/ncp-web/wizard/JS/wizard.js.
 
USBdrive to myCloudDrive

```bash
// Launch nc-datadir
	$('#nc-datadir').on('click', function() {
      launch_action( 'nc-datadir', {"DATADIR":"/media/USBdrive/ncdata"}, nextTabOnSuccess );
});
```
to

```bash
// Launch nc-datadir
	$('#nc-datadir').on('click', function() {
      launch_action( 'nc-datadir', {"DATADIR":"/media/myCloudDrive/ncdata"}, nextTabOnSuccess );
});
```

**Comment by nachoparker** [comment]
It seems that your drive is formatted correctly, but then it doesn't automount? can you verify that? `nc-datadir` is (wisely) refusing to copy the data to the SD card instead of the USB drive

The other one is a NC related problem that I have never seen but it is not an NCP issue

**Comment by Hofer-Julian** [comment]
The drive is formatted correctly and then it also automounts. But as mentioned [here](https://github.com/nextcloud/nextcloudpi/wiki/How-to-configure-an-external-USB-drive-with-NextCloudPi) NextcloudPi comes into troubles when there are more than two drives and the datadir is pointed towards USBdrive instead of the label "myCloudDrive"

**Comment by nachoparker** [comment]
@Hofer-Julian oh I see, you have plugged in two USBdrives even for the wizard. It should work with one, right?

**Comment by Hofer-Julian** [comment]
No, I've one USB drives plugged in for the wizard. The data-dir gets moved to "media/USBdrive/ncdata".

And when I plug in a second drive, the second drive becomes "USBdrive". Of course nextcloud finds no database anymore and produces an error.

But when I skip the wizard and format the drive, activate automount and move datadir to  /media/myCloudDrive/ncdata (not USBdrive) (with your tools) manually, everything works.

Thanks for you fast reply, btw :)

**Comment by prog-amateur** [comment]
> You can always restore your password from SSH or by plugging in a terminal and a keyboard and running `ncp-config` > `nc-passwd`

Hello, does the TUI show the password too, or does it hide it ? I have stopped using NCP because of this inconvenience, but I'm hopeful that somebody has found a solution (I would love to help but I am not a programmer, despite my nickname). 
Thank you !

**Comment by chiyi4488** [comment]
Hey everyone, I have an idea, can we set some fonts for ncp-web, like Noto Sans TC, which supports Chinese and English. Of course, because ncp-web also has other language displays, other fonts are also required.

Why do I have this idea? It is very difficult to display Chinese normally (if it is not installed on a Chinese font)
Even nacho on his website (article), the Chinese screenshots shown are not correct (https://ownyourbits.com/2018/05/29/nextcloudpi-brings-nc13-0-2-automatic- Nc-upgrades-rock64-and-banana-pi-support-armbian-integration-chinese-language-and-more/)

This made me realize that even my own computer is not very good at displaying Chinese. More specifically, it makes me vexed. On the website, we need to focus on the beauty, and I have found Noto Sans TC, which supports Chinese and English perfectly. As for other languages... I don't think it is supported (so I think we should set it up) Multi-font, like switching in the language bar, etc.)

This is my website, you can see Noto Sans TC inside.
https://jmes311chiyi.blogspot.com/

This is the font URL.
https://fonts.google.com/earlyaccess#Noto+Sans+TC
Https://www.google.com/get/noto/help/cjk/
Https://www.google.com/get/noto/

Most of the information is here
https://github.com/nextcloud/server/issues/11640

Thanks.

**Comment by TomTurnschuh** [comment]
@nachoparker: How about creating a GitHub project for this?

**Comment by nachoparker** [comment]
@TomTurnschuh what's that? never used it

**Comment by chiyi4488** [comment]
![image](https://user-images.githubusercontent.com/30853081/50554252-458d0700-0cf2-11e9-9459-7bd3f27a704e.png)

**Comment by TomTurnschuh** [comment]
> 
> 
> @TomTurnschuh what's that? never used it

Me neither. But it might be helpful by providing a structure into such an anhancement project.
A live example can be found in [nextcloud/desktop](https://github.com/nextcloud/desktop/projects/1).

**Comment by nachoparker** [comment]
Well looks good, why not?

There's just so many related little tasks that I wanted to unify them versus having many open issues.

I have been making progress lately with these changes.

**Comment by TomTurnschuh** [comment]
> There's just so many related little tasks that I wanted to unify them versus having many open issues.

Maybe the project's note cards could help with this.

> I have been making progress lately with these changes.

I'm looking forward to using them. :smiley:
